gossipsub.go

```diff
@@ -389,6 +390,24 @@ func WithPeerExchange(doPX bool) Option {
 		}
 
 		gs.doPX = doPX
+		gs.pubPX = false
+
+		return nil
+	}
+}
+
+// WithPeerExchange is a gossipsub router option that enables Peer eXchange on PRUNE.
+// This should generally be enabled in bootstrappers and well connected/trusted nodes
+// used for bootstrapping but using only Signer PeerRecords that include public addreses.
+func WithPublicAddrsOnPeerExchange(doPX bool, pubMaddrsOnly bool) Option {
```

Comment:
I would make this manipulate only `pubPX`, and `WithPeerExchange` shouldn't manipulate `pubPX`. This way there is no order dependency between these options.